### PR TITLE
Fixed heroes column

### PR DESCRIPTION
**Pull request by Argn0**
fixes https://github.com/odota/web/issues/1378

### CODE REVIEW

**Comment by howardchung** [comment]
I'm wondering if we should apply this to all tables (always fix the first column). since this is also a problem on match page tables.  Could we make it part of the Table component?

File: src/components/Table/Table.jsx [hunk 1]
```diff
@@ -112,6 +112,18 @@ class Table extends React.Component {
     this.state = initialState;
   }
 
+  setInnerContainerRef = (node) => { this.innerContainerRef = node; }
+
+  handleScroll = () => {
+    const { scrolled } = this.state;
+    const { scrollLeft } = this.innerContainerRef.childNodes[0].childNodes[0];
```
Comment (howardchung) [inline src/components/Table/Table.jsx]: this seems very specific.  could it fail if the table structure changes?

**Comment by Argn0** [comment]
I found an alternative method. Maybe this is a better solution?

**Comment by howardchung** [comment]
Yes, I think this is better.  Do you have a version where the sponsor button disappears on scroll?  That might actually be OK.

**Comment by Argn0** [comment]
setting the event listener in componentDidUpdate didnt work for the tables below. (ability builds).

So I have to set it in componendDidMount too

**Comment by howardchung** [comment]
Is there a reason to do it in componentDidUpdate?  can't you just do it in mount?

**Comment by Argn0** [comment]
see my discord message.:

ref callbacks only get called before componentDidMount if the element you want set the ref for actually gets rendered. so if there is a loading condition like ```{!loading && <div ref={.....}>...</div>}``` it doesnt work

**Comment by Argn0** [comment]
also see https://stackoverflow.com/a/50019873

**Comment by howardchung** [comment]
But if the component isn't rendering, do we need the table ref to be assigned?  shouldn't the table not be showing in that case?

**Comment by Argn0** [comment]
We need the ref to bind the onscroll event listener to tableDiv. tableDiv is the container of material-ui tables.

**Comment by Argn0** [comment]
Actually forget what I said. It works just fine like this...

**Comment by Argn0** [comment]
I fixed the first column on the large heroes table by default on mobile. I'm worried about bad performance on low tier mobiles.

**Comment by howardchung** [comment]
I think I actually prefer it not to default to fixed, it shouldn't look different unless the user interacts with the table.

**Comment by Argn0** [comment]
undid commit with default scrolled

**Pull request merged**